wireapp/wire-android-reloaded
# feat: (AR-1553 and AR-1649) Splitting search query for federated search + refactor of use cases

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-1553" title="AR-1553" target="_blank"><img alt="Subtask" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10816?size=medium" />AR-1553</a>  Adapt current user search to support federated search (UI)
  </td></table>
  <br />
 










----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues
enables users to search for users in other federations when they search with their id and federation in format i.e "damian.kasperski@wire.com" but when the search query doesn't match regex it's searching only in team you belong

### Causes (Optional)

_Briefly describe the causes behind the issues. This could be helpful to understand the adopted solutions behind some nasty bugs or complex issues._

### Solutions

_Briefly describe the solutions you have implemented for the issues explained above._

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:


### Testing


#### Test Coverage (Optional)


#### How to Test

1. Start new conversation 
2. lookup for "damiankasperski" if you have me in contacts it should show it me grouped in contacts list if now then in "public" group the same should happen for searching "damiankasperski@wire.com" it should show me in "contact" group because we are in contacts
3. for the actual federation search you need to use federated backends anta+bella but overall behavior should be the same if you have someone in contacts it will be shown in contacts otherwise it will be in `public wire` group and to find someone from other federation there is need to provide id and domain
### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
https://wearezeta.atlassian.net/browse/AR-1553
https://wearezeta.atlassian.net/browse/AR-1649

1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
